k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -36,7 +36,9 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
       case (_, Return(rep)) if rep.status.code >= 200 && rep.status.code < 300 => false
       case (_, Return(rep)) if rep.status == http.Status.NotFound || rep.status == http.Status.Gone => false
       // All other status codes are unexpected and should be retried
-      case (_, Return(rep)) => true
+      case (_, Return(rep)) =>
+        log.error("retrying k8s request to %s on unexpected response code %d with message %s", path, rep.statusCode, rep.contentString)
```

Comment:
maybe _warn_ instead of error? let's try to reserve error for things that we _know_ are terrible. this could be something transient if i'm not mistaken.